### PR TITLE
CORTX-32121: PR Build Pipelines 

### PR DESCRIPTION
**Pull request by nitisdev**
# Problem Statement
- Improve cortx-rgw build optimization time in custom-ci and PR jobs.

# Design
-  Added concurrent builds for pr jobs in container.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing

- pr-build: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/pr-build-generic/job/cortx-rgw-pr-build/12/console
- githash fix: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/cortx-rgw-custom-build/1880/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shailesh-vaidya — COMMENTED** [review]
Looks good to me.  Can you please try running it as a non-mandatory check for cortx-rgw PRs

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 1]
```diff
@@ -64,53 +67,53 @@ pipeline {
...
+                '''
             }
         }
         // Release cortx deployment stack
         stage('Release') {
             steps {
                 script { build_stage = env.STAGE_NAME }
                 dir('cortx-re') {
-                    checkout([$class: 'GitSCM', branches: [[name: '*/main']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', depth: 1, honorRefspec: true, noTags: true, reference: '', shallow: true], [$class: 'AuthorInChangelog']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: 'https://github.com/Seagate/cortx-re']]])
+                    checkout([$class: 'GitSCM', branches: [[name: '*/CORTX_32121_pr']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', depth: 1, honorRefspec: true, noTags: true, reference: '', shallow: true], [$class: 'AuthorInChangelog']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: 'https://github.com/nitisdev/cortx-re']]])
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:116]: please change branch and repo to main.

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 2]
```diff
@@ -168,7 +171,7 @@ pipeline {
                 script { build_stage = env.STAGE_NAME }
                 script {
                     try {
-                        def buildCortxAllImage = build job: 'cortx-docker-images-for-PR', wait: true,
+                        def buildCortxAllImage = build job: '/Cortx-PR-Build/Cortx-Deployment/Generic/cortx-docker-images-for-PR', wait: true,
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:174]: this path will change as well after successful test

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 3]
```diff
@@ -195,7 +198,7 @@ pipeline {
             steps {
                 script { build_stage = env.STAGE_NAME }
                 script {
-                    build job: "K8s-1N-deployment", wait: true,
+                    build job: "/Cortx-PR-Build/Cortx-Deployment/Generic/K8s-1N-deployment", wait: true,
```
Comment (gauravchaudhari02) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:201]: this path will change as well after successful test

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 1 above)
Comment (nitisdev) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy]: Done.

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 2 above)
Comment (nitisdev) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy]: Done

File: solutions/kubernetes/ceph-binary-build.sh [hunk 4]
```diff
@@ -185,7 +186,8 @@ function ceph_build() {
                 dnf config-manager --set-enabled powertools
                 pushd "$BUILD_LOCATION"
                     add_common_separator "Clone Repo"
-                    git clone $CEPH_REPO -b $CEPH_BRANCH
+                    git clone $CEPH_REPO
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-binary-build.sh:189]: Use `git clone $CEPH_REPO && git checkout $CEPH_BRANCH`

File: solutions/kubernetes/ceph-binary-build.sh [hunk 5]
```diff
@@ -236,7 +238,8 @@ function ceph_build() {
...
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-binary-build.sh:241]: Use `git clone $CEPH_REPO && git checkout $CEPH_BRANCH`

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: solutions/kubernetes/ceph-binary-build.sh (hunk 4 above)
Comment (nitisdev) [reply solutions/kubernetes/ceph-binary-build.sh]: Done

File: solutions/kubernetes/ceph-binary-build.sh (hunk 5 above)
Comment (nitisdev) [reply solutions/kubernetes/ceph-binary-build.sh]: Done

**Pull request merged**